apps/console/src/features/roles/components/wizard-updated/application-role-wizard.tsx

```diff
@@ -386,49 +392,69 @@ export const ApplicationRoleWizard: FunctionComponent<ApplicationRoleWizardProps
+                            paper: {
+                                elevation: 2
+                            },
+                            popper: {
+                                modifiers: [
+                                    {
+                                        enabled: false,
+                                        name: "flip"
+                                    },
+                                    {
+                                        enabled: false,
+                                        name: "preventOverflow"
+                                    }
+                                ]
+                            }
+                        } }
+                        data-componentid={ `${componentId}-api` }
+                        getOptionLabel={ (apiResourcesListOption: DropdownProps) =>
+                            apiResourcesListOption.text }
+                        groupBy={ (apiResourcesListOption: DropdownItemProps) =>
+                            APIResourceUtils
+                                .resolveApiResourceGroup(apiResourcesListOption?.type) }
+                        isOptionEqualToValue={
+                            (option: DropdownItemProps, value: DropdownItemProps) =>
+                                option.value === value.value
                         }
-                        loading={ false }
-                        onChange={ onAPIResourceSelected }
-                        hint={
-                            !isSubscribedAPIResourcesListLoading
-                            && apiResourcesListOptions?.length === 0
-                            && selectedAPIResources?.length === 0
-                            && (
-                                <Hint>
-                                    <Trans
-                                        i18nKey={ "console:manage.features.roles.addRoleWizard.forms.rolePermission" +
-                                            ".apiResource.hint.empty" }
-                                    >
-                                        There are no API resources authorized for the selected application.
-                                        API Resources can be authorized through <Link
-                                            external={ false }
-                                            onClick={ () => {
-                                                history.push(
-                                                    `${ AppConstants.getPaths()
-                                                        .get("APPLICATION_EDIT").replace(":id", appId) }#tab=4`
-                                                );
-                                            } }
-                                        > here </Link>
-                                    </Trans>
-                                </Hint>
+                        options={
+                            apiResourcesListOptions?.filter((item: DropdownProps) =>
```

Comment:
When `apiResourcesListOptions` is not populated at the time of rendering, this can break the whole page. It'd be better to conditionally render the autocomplete input based on the availability of `apiResourcesListOptions`

Reply:
Fixed